delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/Resources.scala

```diff
@@ -267,40 +280,24 @@ object Resources {
 
   @SuppressWarnings(Array("OptionGet"))
   private[delta] def evaluate(
-      resourceResolution: ResourceResolution[Schema]
+      resourceValidator: ResourceValidator
   )(state: Option[ResourceState], cmd: ResourceCommand)(implicit
-      api: JsonLdApi,
       clock: Clock[UIO]
   ): IO[ResourceRejection, ResourceEvent] = {
 
-    def toGraph(id: Iri, expanded: ExpandedJsonLd): IO[ResourceRejection, Graph] =
-      IO.fromEither(expanded.toGraph).mapError(err => InvalidJsonLdFormat(Some(id), err))
-
     def validate(
         projectRef: ProjectRef,
         schemaRef: ResourceRef,
         caller: Caller,
         id: Iri,
         expanded: ExpandedJsonLd
-    ): IO[ResourceRejection, (ResourceRef.Revision, ProjectRef)] =
-      if (schemaRef == Latest(schemas.resources) || schemaRef == ResourceRef.Revision(schemas.resources, 1))
-        IO.raiseWhen(id.startsWith(contexts.base))(ReservedResourceId(id)) >>
-          toGraph(id, expanded) >>
-          IO.pure((ResourceRef.Revision(schemas.resources, 1), projectRef))
-      else
-        for {
-          _        <- IO.raiseWhen(id.startsWith(contexts.base))(ReservedResourceId(id))
-          graph    <- toGraph(id, expanded)
-          resolved <- resourceResolution
-                        .resolve(schemaRef, projectRef)(caller)
-                        .mapError(InvalidSchemaRejection(schemaRef, projectRef, _))
-          schema   <-
-            if (resolved.deprecated) IO.raiseError(SchemaIsDeprecated(resolved.value.id)) else IO.pure(resolved)
-          dataGraph = graph ++ schema.value.ontologies
-          report   <- ShaclEngine(dataGraph, schema.value.shapes, reportDetails = true, validateShapes = false)
-                        .mapError(ResourceShaclEngineRejection(id, schemaRef, _))
-          _        <- IO.when(!report.isValid())(IO.raiseError(InvalidResource(id, schemaRef, report, expanded)))
-        } yield (ResourceRef.Revision(schema.id, schema.rev), schema.value.project)
+    ): IO[ResourceRejection, (ResourceRef.Revision, ProjectRef)] = {
+      for {
+        (_, rev, ref) <- resourceValidator.validationReport(projectRef, schemaRef, caller, id, expanded)
+      } yield {
+        (rev, ref)
+      }
+    }
```

Comment:
This code here is why it would be nicer to have two validation methods for the two different use cases